Inferences/ForwardDemodulation.cpp

```diff
@@ -116,45 +131,26 @@ bool ForwardDemodulationImpl<combinatorySupSupport>::perform(Clause* cl, Clause*
           continue;
         }
 
+        auto lhs = qr.data->term;
+
         // TODO:
         // to deal with polymorphic matching
         // Ideally, we would like to extend the substitution
         // returned by the index to carry out the sort match.
         // However, ForwardDemodulation uses a CodeTree as its
         // indexing mechanism, and it is not clear how to extend
         // the substitution returned by a code tree.
-        Recycled<RobSubstitution> subst;
```

Comment:
Turns out this uses a lot (up to 5%) of the runtime. Does anyone have any idea what causes it? Maybe called too many times?

Reply:
`Recycled<T>` has to do some reset logic and lookup some recycled object, whereas a static will just do some initialisation logic (at least with `-fno-threadsafe-statics`, which we have). So `Recycled<T>` will be slower, but may re-use more objects.

Reply:
Okay, so can we have this for now? Or should I revert?

Reply:
Yes, of course - I was just explaining the rationale, I think your change is pragmatic.